server/node/src/main/java/org/kaaproject/kaa/server/admin/client/mvp/activity/EndpointProfilesActivity.java

```diff
@@ -88,6 +89,17 @@ public void onRowAction(RowActionEvent<String> event) {
                 String id = event.getClickedId();
                 if (event.getAction()==RowActionEvent.CLICK) {
                     goTo(new EndpointProfilePlace(applicationId, id, gridLoaded));
+                } else if (event.getAction()==RowActionEvent.DELETE) {
+                    deleteEntity(id, new BusyAsyncCallback<Void>() {
+                        @Override
+                        public void onFailureImpl(Throwable caught) {
+                            Utils.handleException(caught, listView);
+                        }
+
+                        @Override
+                        public void onSuccessImpl(Void result) {}
+                    });
+                    dataProvider.update();
                 }
```

Comment:
deleteEntity is asynchronous operation and dataProvider.update() must be called after successful deletion (in success onSuccessImpl callback).